pkg/service/types.go

```diff
@@ -43,9 +45,21 @@ type MeshService struct {
 
 	// Protocol is the protocol served by the service's port
 	Protocol string
+
+	// ProviderKey represents the name of the original entity from which this MeshService was created (e.g. a Kubernetes service)
+	ProviderKey string
+
+	// Subdomain is an optional subdomain for this MeshService
+	Subdomain string
```

Comment:
can you give an example of when we use subdomain?

Reply:
[Endpoint Listing](https://github.com/openservicemesh/osm/pull/4704/files#diff-a3fcf30b91ac74a293c1b1796ebb8b53d9f8bf748472a72a462607cadaeb6d55R58) and [FQDN](https://github.com/openservicemesh/osm/pull/4704/files#diff-428786751bb8848d286e65326ee03834f68486a3897130242ea3175534490389R79) are the two main ones